consensus/consensus_v2.go

```diff
@@ -679,13 +680,66 @@ func (consensus *Consensus) commitBlock(blk *types.Block, committedMsg *FBFTMess
 	return nil
 }
 
+func (consensus *Consensus) rotateLeader(epoch *big.Int) {
+	prev := consensus.GetLeaderPubKey()
+	curNumber := consensus.Blockchain.CurrentHeader().Number().Uint64()
+	utils.Logger().Info().Msgf("[Rotating leader] epoch: %v rotation:%v numblocks:%d", epoch.Uint64(), consensus.Blockchain.Config().IsLeaderRotation(epoch), consensus.Blockchain.Config().LeaderRotationBlocksCount)
+	if consensus.Blockchain.Config().IsLeaderRotation(epoch) {
+		leader := consensus.GetLeaderPubKey()
+		for i := 0; i < consensus.Blockchain.Config().LeaderRotationBlocksCount; i++ {
+			header := consensus.Blockchain.GetHeaderByNumber(curNumber - uint64(i))
+			if header == nil {
+				return
+			}
+			// Previous epoch, we should not change leader.
```

Comment:
Nit: Previous `block was epoch block`, we should not change leader.

Reply:
fixed